phantomgames/Home/HomeScreenViewController.swift

```diff
@@ -33,51 +43,77 @@ class HomeScreenViewController: UIViewController{
+    }
+    @objc private func pcImageTapped() {
+        navigateToSearchGameScreen(with: Constants.Endpoints.pcGamesURL)
+    }
+    
+    @objc private func browserImageTapped() {
+        navigateToSearchGameScreen(with: Constants.Endpoints.browserGamesURL)
+    }
+    
+    private func navigateToSearchGameScreen(with url: String) {
+        let storyboard = UIStoryboard(name: "SearchGame", bundle: nil)
+        if let searchGameVC = storyboard.instantiateViewController(withIdentifier: "SearchGameViewController") as? SearchGameViewController {
+            searchGameVC.gamesURL = url
+            navigationController?.pushViewController(searchGameVC, animated: true)
+        }
+    }
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == Constants.SegueIdentifiers.GameDetailScreenSegue,
+           let destinationVC = segue.destination as? GameDetailViewController,
+           let gameID = sender as? Int {
+            destinationVC.assignGameID(gameID: gameID)
+        }
+    }
 }
 
-// MARK:  Collection View
+// MARK: Collection View
 
-extension HomeScreenViewController: UICollectionViewDataSource,UICollectionViewDelegate,UICollectionViewDelegateFlowLayout {
+extension HomeScreenViewController: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
     public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        viewModel.allGameList.count
+        viewModel.collectionViewGamesCount
     }
     
     public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as!CustomCollectionViewCell
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as? CustomCollectionViewCell else {
+            return UICollectionViewCell()
+        }
         
-        if let game = viewModel.game(atIndex: indexPath.item) {
-            cell.ConfigCellWith(game: game)
+        if let game = viewModel.collectionViewGame(atIndex: indexPath.item) {
+            cell.configCellWith(game: game)
         }
         return cell
     }
     
     public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath:IndexPath) {
         guard let cell = collectionView.cellForItem(at: indexPath) as? CustomCollectionViewCell,
-              let gameID = viewModel.game(atIndex: indexPath.item)?.gameID else {
+              let gameID = viewModel.collectionViewGame(atIndex: indexPath.item)?.gameID else {
             displayAlert(title: "Error", message: "Failed to select game. Please try again.", buttonTitle: "OK")
             return
         }
         self.performSegue(withIdentifier: Constants.SegueIdentifiers.GameDetailScreenSegue, sender: gameID)
```

Comment:
I know this isn't part of the current changes, but since you are touching the file just check, you should one able to remove self here
```suggestion
        performSegue(withIdentifier: Constants.SegueIdentifiers.GameDetailScreenSegue, sender: gameID)
```